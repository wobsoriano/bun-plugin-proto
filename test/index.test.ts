import { expect, test } from 'bun:test'
import protoPlugin from '../src'

Bun.plugin(protoPlugin())

test('generate result given message proto file', async () => {
  const { default: proto } = await import('./user.proto')
  expect(proto).toMatchSnapshot()
})
