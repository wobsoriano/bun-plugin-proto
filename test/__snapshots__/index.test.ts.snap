// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate result given message proto file 1`] = `
{
  "User": {
    "fileDescriptorProtos": [
      {
        "data": [
          10,
          10,
          114,
          111,
          111,
          116,
          46,
          112,
          114,
          111,
          116,
          111,
          34,
          173,
          1,
          10,
          4,
          85,
          115,
          101,
          114,
          18,
          17,
          10,
          9,
          102,
          105,
          114,
          115,
          116,
          78,
          97,
          109,
          101,
          24,
          1,
          32,
          1,
          40,
          9,
          18,
          16,
          10,
          8,
          108,
          97,
          115,
          116,
          78,
          97,
          109,
          101,
          24,
          2,
          32,
          1,
          40,
          9,
          18,
          14,
          10,
          6,
          97,
          99,
          116,
          105,
          118,
          101,
          24,
          3,
          32,
          1,
          40,
          8,
          18,
          21,
          10,
          7,
          109,
          97,
          110,
          97,
          103,
          101,
          114,
          24,
          4,
          32,
          1,
          40,
          11,
          50,
          4,
          85,
          115,
          101,
          114,
          18,
          17,
          10,
          9,
          108,
          111,
          99,
          97,
          116,
          105,
          111,
          110,
          115,
          24,
          5,
          32,
          3,
          40,
          9,
          18,
          26,
          10,
          8,
          112,
          114,
          111,
          106,
          101,
          99,
          116,
          115,
          24,
          6,
          32,
          3,
          40,
          11,
          50,
          8,
          80,
          114,
          111,
          106,
          101,
          99,
          116,
          115,
          26,
          42,
          10,
          8,
          80,
          114,
          111,
          106,
          101,
          99,
          116,
          115,
          18,
          11,
          10,
          3,
          107,
          101,
          121,
          24,
          1,
          32,
          1,
          40,
          9,
          18,
          13,
          10,
          5,
          118,
          97,
          108,
          117,
          101,
          24,
          2,
          32,
          1,
          40,
          9,
          58,
          2,
          56,
          1,
          98,
          6,
          112,
          114,
          111,
          116,
          111,
          51,
        ],
        "type": "Buffer",
      },
    ],
    "format": "Protocol Buffer 3 DescriptorProto",
    "type": {
      "enumType": [],
      "extension": [],
      "extensionRange": [],
      "field": [
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "firstName",
          "number": 1,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_STRING",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "lastName",
          "number": 2,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_STRING",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "active",
          "number": 3,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_BOOL",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "manager",
          "number": 4,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_MESSAGE",
          "typeName": "User",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_REPEATED",
          "name": "locations",
          "number": 5,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_STRING",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_REPEATED",
          "name": "projects",
          "number": 6,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_MESSAGE",
          "typeName": "Projects",
        },
      ],
      "name": "User",
      "nestedType": [
        {
          "enumType": [],
          "extension": [],
          "extensionRange": [],
          "field": [
            {
              "defaultValue": "",
              "extendee": "",
              "jsonName": "",
              "label": "LABEL_OPTIONAL",
              "name": "key",
              "number": 1,
              "oneofIndex": 0,
              "options": null,
              "type": "TYPE_STRING",
              "typeName": "",
            },
            {
              "defaultValue": "",
              "extendee": "",
              "jsonName": "",
              "label": "LABEL_OPTIONAL",
              "name": "value",
              "number": 2,
              "oneofIndex": 0,
              "options": null,
              "type": "TYPE_STRING",
              "typeName": "",
            },
          ],
          "name": "Projects",
          "nestedType": [],
          "oneofDecl": [],
          "options": {
            "deprecated": false,
            "mapEntry": true,
            "messageSetWireFormat": false,
            "noStandardDescriptorAccessor": false,
            "uninterpretedOption": [],
          },
          "reservedName": [],
          "reservedRange": [],
        },
      ],
      "oneofDecl": [],
      "options": null,
      "reservedName": [],
      "reservedRange": [],
    },
  },
}
`;

exports[`generate result given message proto file 1`] = `
{
  "User": {
    "fileDescriptorProtos": [
      {
        "data": [
          10,
          10,
          114,
          111,
          111,
          116,
          46,
          112,
          114,
          111,
          116,
          111,
          34,
          173,
          1,
          10,
          4,
          85,
          115,
          101,
          114,
          18,
          17,
          10,
          9,
          102,
          105,
          114,
          115,
          116,
          78,
          97,
          109,
          101,
          24,
          1,
          32,
          1,
          40,
          9,
          18,
          16,
          10,
          8,
          108,
          97,
          115,
          116,
          78,
          97,
          109,
          101,
          24,
          2,
          32,
          1,
          40,
          9,
          18,
          14,
          10,
          6,
          97,
          99,
          116,
          105,
          118,
          101,
          24,
          3,
          32,
          1,
          40,
          8,
          18,
          21,
          10,
          7,
          109,
          97,
          110,
          97,
          103,
          101,
          114,
          24,
          4,
          32,
          1,
          40,
          11,
          50,
          4,
          85,
          115,
          101,
          114,
          18,
          17,
          10,
          9,
          108,
          111,
          99,
          97,
          116,
          105,
          111,
          110,
          115,
          24,
          5,
          32,
          3,
          40,
          9,
          18,
          26,
          10,
          8,
          112,
          114,
          111,
          106,
          101,
          99,
          116,
          115,
          24,
          6,
          32,
          3,
          40,
          11,
          50,
          8,
          80,
          114,
          111,
          106,
          101,
          99,
          116,
          115,
          26,
          42,
          10,
          8,
          80,
          114,
          111,
          106,
          101,
          99,
          116,
          115,
          18,
          11,
          10,
          3,
          107,
          101,
          121,
          24,
          1,
          32,
          1,
          40,
          9,
          18,
          13,
          10,
          5,
          118,
          97,
          108,
          117,
          101,
          24,
          2,
          32,
          1,
          40,
          9,
          58,
          2,
          56,
          1,
          98,
          6,
          112,
          114,
          111,
          116,
          111,
          51,
        ],
        "type": "Buffer",
      },
    ],
    "format": "Protocol Buffer 3 DescriptorProto",
    "type": {
      "enumType": [],
      "extension": [],
      "extensionRange": [],
      "field": [
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "firstName",
          "number": 1,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_STRING",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "lastName",
          "number": 2,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_STRING",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "active",
          "number": 3,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_BOOL",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_OPTIONAL",
          "name": "manager",
          "number": 4,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_MESSAGE",
          "typeName": "User",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_REPEATED",
          "name": "locations",
          "number": 5,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_STRING",
          "typeName": "",
        },
        {
          "defaultValue": "",
          "extendee": "",
          "jsonName": "",
          "label": "LABEL_REPEATED",
          "name": "projects",
          "number": 6,
          "oneofIndex": 0,
          "options": null,
          "type": "TYPE_MESSAGE",
          "typeName": "Projects",
        },
      ],
      "name": "User",
      "nestedType": [
        {
          "enumType": [],
          "extension": [],
          "extensionRange": [],
          "field": [
            {
              "defaultValue": "",
              "extendee": "",
              "jsonName": "",
              "label": "LABEL_OPTIONAL",
              "name": "key",
              "number": 1,
              "oneofIndex": 0,
              "options": null,
              "type": "TYPE_STRING",
              "typeName": "",
            },
            {
              "defaultValue": "",
              "extendee": "",
              "jsonName": "",
              "label": "LABEL_OPTIONAL",
              "name": "value",
              "number": 2,
              "oneofIndex": 0,
              "options": null,
              "type": "TYPE_STRING",
              "typeName": "",
            },
          ],
          "name": "Projects",
          "nestedType": [],
          "oneofDecl": [],
          "options": {
            "deprecated": false,
            "mapEntry": true,
            "messageSetWireFormat": false,
            "noStandardDescriptorAccessor": false,
            "uninterpretedOption": [],
          },
          "reservedName": [],
          "reservedRange": [],
        },
      ],
      "oneofDecl": [],
      "options": null,
      "reservedName": [],
      "reservedRange": [],
    },
  },
}
`;
